### PR TITLE
Add Helm Chart to apache-airflow docs

### LINKED ISSUE #10523: Host Airflow-managed Helm repo
**Description**

@mik-laj Hi 👋 I was just coming to this repo to see if you're interested in help getting the Helm chart repo set up to replace the chart in `stable`. 

Stable repo is nearing the end of it's deprecation period, and I'm glad to see a version of the Airflow chart is already here. See https://github.com/helm/charts/issues/21103 for the meta issue tracking moving all the `stable` repo charts to new homes.

### To-do

- [ ] Decide on hosting options below (self-host/leverage Bitnami)
- [ ] If self-host, set up CI/CD for chart-testing and chart-releasing
- [ ] if self-host, list in Helm Hub (http://hub.helm.sh/) and Artifact Hub (https://artifacthub.io/)

**Use case / motivation**

### Set up Helm repo hosting for your chart

Set up a Helm repo, either as a separate git repo in your org, or keeping the same setup you have now. We have created Helm chart repo actions for chart testing (CI) and releasing chart packages as your own GitHub-hosted Helm repo (CD).

#### Self-hosted options:
1. If we either move the chart to a separate git repo in the artifacthub gh org, or even move the hub github pages setting to a branch other than the main one, we can use the [@helm/chart-releaser-action](https://github.com/helm/chart-releaser-action) GitHub Action to automate the helm repo.
2. If we keep structure as-is, we can still use the [helm/chart-releaser](https://github.com/helm/chart-releaser) project, just with a custom script.

For either option we can also use the [@helm/chart-testing-action](https://github.com/helm/chart-testing-action) to wrap the chart-testing project @mattfarina mentioned above. Here's an demo repo to see how they work together: https://github.com/helm/charts-repo-actions-demo

Whichever option you decide I'm make a PR if it helps.

If you do decide to host your own Helm repo, you will also want to list it in 

#### Alternatively leverage existing Bitnami Helm repo

There is also a version of the chart maintained by Bitnami, who have been very involved in the `stable` repo for years, but : https://github.com/bitnami/charts/tree/master/bitnami/airflow. You could instead decide to leverage that chart as the canonical source, and not host your own. It is also fine to have multiple instances of a chart to install the same app.

**Related Issues**

- https://github.com/helm/charts/issues/21103
- https://github.com/apache/airflow/issues/10486
- https://github.com/apache/airflow/issues/10379

### PR DESCRIPTION
**Pull request by mik-laj**
Based on: https://github.com/apache/airflow/pull/12755 

Close: https://github.com/apache/airflow/issues/10523

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: docs/apache-airflow/kubernetes.rst [hunk 1]
```diff
@@ -20,6 +20,15 @@
...
 
+Apache Airflow aims to be a very Kubernetes-friendly project, and many users run Airflow
+from within a Kubernetes cluster in order to take advantage of the increased stability
+and autoscaling options that Kubernetes provides.
+
+Helm Chart for Kubernetes
+=========================
+
+We maintain :doc:`official Helm chart <helm-chart:index>` for Airflow that help you define, install, and upgrade deployment.
+
```
Comment (kaxil) [inline docs/apache-airflow/kubernetes.rst]:
```suggestion
We maintain :doc:`official Helm chart <helm-chart:index>` for Airflow that helps you define, install, and upgrade deployment.

```

File: docs/apache-airflow/production-deployment.rst [hunk 2]
```diff
@@ -120,6 +120,12 @@ Production Container Images
 
 We provide :doc:`a Docker Image (OCI) for Apache Airflow <docker-stack:index>` for use in a containerized environment. Consider using it to guarantee that software will always run the same no matter where it’s deployed.
 
+Helm Chart for Kubernetes
+=========================
+
+`The Helm <https://helm.sh/>`__ provides a simple mechanism to deploy software to a Kubernetes cluster. We maintain
+:doc:`official Helm chart <helm-chart:index>` for Airflow that help you define, install, and upgrade deployment.
```
Comment (kaxil) [inline docs/apache-airflow/production-deployment.rst]:
```suggestion
:doc:`official Helm chart <helm-chart:index>` for Airflow that helps you define, install, and upgrade deployment.
```

**Review by kaxil — APPROVED** [review]
Minor suggestions

File: docs/apache-airflow/kubernetes.rst [hunk 3]
```diff
@@ -20,6 +20,15 @@
...
 ----------
 
+Apache Airflow aims to be a very Kubernetes-friendly project, and many users run Airflow
+from within a Kubernetes cluster in order to take advantage of the increased stability
+and autoscaling options that Kubernetes provides.
+
+Helm Chart for Kubernetes
+=========================
+
+We maintain :doc:`official Helm chart <helm-chart:index>` for Airflow that helps you define, install, and upgrade deployment.
```
Comment (potiuk) [inline docs/apache-airflow/kubernetes.rst]:
```suggestion
We maintain :doc:`official Helm chart <helm-chart:index>` for Airflow that helps you define, install, and upgrade deployment. The Helm Chart uses :doc:`official Docker image and Dockerfile <docker-stack:index>` that is also maintained and released by the community.
```

File: docs/apache-airflow/production-deployment.rst [hunk 4]
```diff
@@ -120,6 +120,12 @@ Production Container Images
 
 We provide :doc:`a Docker Image (OCI) for Apache Airflow <docker-stack:index>` for use in a containerized environment. Consider using it to guarantee that software will always run the same no matter where it’s deployed.
 
+Helm Chart for Kubernetes
+=========================
+
+`The Helm <https://helm.sh/>`__ provides a simple mechanism to deploy software to a Kubernetes cluster. We maintain
+:doc:`official Helm chart <helm-chart:index>` for Airflow that helps you define, install, and upgrade deployment.
```
Comment (potiuk) [inline docs/apache-airflow/production-deployment.rst]:
```suggestion
:doc:`official Helm chart <helm-chart:index>` for Airflow that helps you define, install, and upgrade deployment. The Helm Chart uses :doc:`official Docker image and Dockerfile <docker-stack:index>` that is also maintained and released by the community.
```

**Review by potiuk — CHANGES_REQUESTED** [review]
I think we should also mention the image which is the foundation for the Helm Chart.

**Comment by mik-laj** [comment]
@potiuk Added.

**Review by potiuk — APPROVED** [review]

**Pull request merged**